modules/mobile/app/services/mobile/v0/legacy_disability_rating/proxy.rb

```diff
@@ -28,17 +28,13 @@ def get_disability_ratings # rubocop:disable Metrics/MethodLength
         rescue EVSS::DisabilityCompensationForm::ServiceUnavailableException
           raise Common::Exceptions::BackendServiceException, 'MOBL_502_upstream_error'
         rescue => e
-          if e.respond_to?('response')
-            case e.response[:status]
-            when 400
-              raise Common::Exceptions::BackendServiceException, 'MOBL_404_rating_not_found'
-            when 502
-              raise Common::Exceptions::BackendServiceException, 'MOBL_502_upstream_error'
-            when 403
-              raise Common::Exceptions::BackendServiceException, 'MOBL_403_rating_forbidden'
-            else
-              raise e
-            end
+          case e&.response&.dig(:status)
```

Comment:
Same issue with the first `&` here. But also, I believe that Jason added the `e.respond_to?('response')` check because we ran into some issue where sometimes the error did not have a `response` method. That doesn't imply that the error is nil (which is probably impossible), just that it might not be a faraday error or other error that has a `response` method, which is not a universal error method. I think the underlying problem here is that this is a blanket rescue (not rescuing a specific error class) so we don't know how that error will be structured. It will sometimes be structured as a faraday error and sometimes not. So I think you still have to ensure that the error has a `response` method.

An alternative approach would probably be to figure out which error type we expect to have a response method (my guess is `Faraday::Error` but I can't be certain) and rescue it separately, then have this blanket rescue. But given that this is legacy code, I doubt it's worth the trouble.

Reply:
Yes, I think you're right that there's another type of error that ideally would be caught separately but also agree that it's not worth the trouble since this is soon to be discontinued. This change will at least prevent an ugly `no [] method for nil object` and raise the uncaught error. The new LH disability rating doesn't have these long case statements for error raising.

Reply:
I think I said too much and confused things. I believe that the code as it's written re-introduces a bug when the error isn't from faraday. I think this needs to be something more like:
```
raise e unless e.respond_to?('response')

case e.response&.dig(:status)
```

Reply:
oh I see it now. I was forgetting that the safety operator only protects against nil. Now updated.